.github/workflows/reusable-cibuildwheel.yml

```
@@ -81,9 +81,11 @@ jobs:
           with-cython-tracing=${{ inputs.cython-tracing }}
 
     - name: Upload built artifacts for testing and publishing
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       with:
-        name: ${{ inputs.dists-artifact-name }}
+        name: ${{ inputs.dists-artifact-name }}-
+          ${{ inputs.os }}-
+          ${{ inputs.qemu }}
```

It might be useful to take this input into account, as #562 revealed the need to run this workflow several times to split wheel building into smaller portions.
```suggestion
          ${{ inputs.qemu }}-
          ${{ inputs.wheel-tags-to-skip }}
```

In the future, though, I'd look into rewriting this to use something like a hash of the entire `${{ inputs }}` object. But that's out of the scope right now.